istioctl/cmd/root.go

```diff
@@ -313,3 +356,14 @@ func unsupportedCmd(cmd *cobra.Command, alternative string) *cobra.Command {
 		},
 	}
 }
+
+// disableUnsupportedCmds is used to disable all the commands that are not supported.
+func disableUnsupportedCmds(cmd *cobra.Command) {
+	unsupportedCmds := buildUnsupportedCmds(UnsupportedCmds)
+	for _, c := range cmd.Commands() {
+		if unsupportedCmds[c.Name()] {
+			cmd.RemoveCommand(c)
+			cmd.AddCommand(unsupportedCmd(c, UnsupportedCmdsMsg, buildUnsupportedCmdsAlternativeMsg(UnsupportedCmdsAlternativeMsg)[c.Name()]))
```

Comment:
I don't think we should be re-building the map of alternative messages for each iteration. How about merging both maps together - each unsupported cmd will have either default message or alternative if specified. WDYT?

Reply:
After refactoring, the `disabledCommands` struct contains all the `disableCommand` with their `name`, `disabled` and `alternative` fields.